### PR TITLE
Middlewares: add forwardAuth.authResponseHeadersRegex

### LINKED ISSUE #6823: ForwardAuth authResponseHeaders accept regexp or wildcard
### Do you want to request a *feature* or report a *bug*?

Feature for `ForwardAuth` middleware to accept in `authResponseHeaders` regular expressions or wildcard

### What did you expect to see?

Real world use-case is adding new headers by some pattern from auth-service without redeploying traefik. For example we could have `X-User-Id` header, but then add `X-User-Role` header. Now we should change current config, but with this feature we could provide `X-User-.*` or `X-User-*` pattern instead at first time.

### PR DESCRIPTION
**Pull request by iamolegga**
### What does this PR do?

adds new field `authResponseHeadersRegex` to `forwardAuth` middleware

### Motivation

fixes #6823 

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

update of outdated pr #6828

### CODE REVIEW

**Comment by iamolegga** [comment]
That's pretty strange for me why not related test has failed, me we should огые trigger check again 🤔

**Comment by jbdoumenjou** [comment]
Hi @iamolegga ,

Thanks for (re)opening the PR. :wink: 
Don't worry about the CI, it is related to another issue that will be fixed with #7436 .

**Review by rtribotte — CHANGES_REQUESTED** [review]
Hello @iamolegga,

Thanks for your work 👍 
I have made several comments.
Also, could you:
- add the new field in the dynamic references in the doc contents (`file.toml`, `file.yml`, `marathon-labels.json` and `docker-labels.yml`) and generate afterward the KV references, trough a `go generate` ?
- add the new field to the [middleware CRD](https://github.com/traefik/traefik/blob/master/pkg/provider/kubernetes/crd/traefik/v1alpha1/middleware.go#L87) ?

File: pkg/middlewares/auth/forward.go [hunk 1]
```diff
@@ -66,6 +68,14 @@ func NewForward(ctx context.Context, next http.Handler, config dynamic.ForwardAu
 		fa.client.Transport = tr
 	}
 
+	if config.AuthResponseHeadersRegex != "" {
+		re, err := regexp.Compile(config.AuthResponseHeadersRegex)
+		if err != nil {
+			return nil, err
```
[the remark this answers is not in the document]
Comment (iamolegga) [reply pkg/middlewares/auth/forward.go]: @rtribotte updated, please check again

**Comment by iamolegga** [comment]
@rtribotte I've updated PR, I hope I didn't miss anything 😀

File: pkg/middlewares/auth/forward.go [hunk 2]
```diff
@@ -66,6 +69,14 @@ func NewForward(ctx context.Context, next http.Handler, config dynamic.ForwardAu
 		fa.client.Transport = tr
 	}
 
+	if config.AuthResponseHeadersRegex != "" {
+		re, err := regexp.Compile(config.AuthResponseHeadersRegex)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile 'forwardAuth.AuthResponseHeadersRegex' value '%s': %w", config.AuthResponseHeadersRegex, err)
```
Comment (rtribotte) [inline pkg/middlewares/auth/forward.go:75]: Thanks for you modifications ;-)
I'm not sure here, i would go this (like in other middlewares)

```suggestion
			return nil, fmt.Errorf("error compiling regular expression %s: %w", config.AuthResponseHeadersRegex, err)
```

**Comment by iamolegga** [comment]
@rtribotte sorry for disturbing, is it correct now?

**Comment by rtribotte** [comment]
Hello @iamolegga,
Thanks for your modifications !
Sorry for the delay, i was awaiting for the update of the documentation references, but it was definitely not clear enough by stating it in my comment.
The idea is to complete the documentation :
 - add the new field in the dynamic references in the doc contents (file.toml, file.yml, marathon-labels.json and docker-labels.yml) 
 - generate afterward the KV references, trough a go generate 
 - add the new field to the middleware CRD 

Don't hesitate if something is not clear ! ;-)

**Comment by iamolegga** [comment]
@rtribotte thanks, finally understood what need to be done. updated 🙂

File: docs/content/middlewares/forwardauth.md [hunk 3]
```diff
@@ -217,6 +217,57 @@ http:
           - "X-Secret"
 ```
 
+### `authResponseHeadersRegex`
+
+The `authResponseHeadersRegex` option is the regex to match the headers that should be copied from the authentication server to the request. All incoming request's headers matching this regex are deleted from the request before any copy happens.
```
Comment (rtribotte) [inline docs/content/middlewares/forwardauth.md:222]: I think it's better to be specific on how the matching behave:

```suggestion
The `authResponseHeadersRegex` option is the regex to match the headers that should be copied from the authentication server to the request. All incoming request's headers matching this regex are deleted from the request before any copy happens.
It allows partial matching of the regular expression in the `accessControlAllowOriginList`.
You should use start of string (`^`) and end of string (`$`) anchors to ensure a full match against the header key.
```

Comment (iamolegga) [reply docs/content/middlewares/forwardauth.md]: Sure, commited 👍

**Review by rtribotte — APPROVED** [review]
LGTM 👍

**Review by jbdoumenjou — APPROVED** [review]
LGTM

**Review by kevinpollet — APPROVED** [review]
LGTM

**Pull request merged**